File: 백준/Silver/1764. 듣보잡/듣보잡.py
dic={}
n,m=map(int, input().split())
cnt=0
li=[]

for _ in range(n):
    l=input()
    dic[l]=1
for _ in range(m):
    s=input()
    if s in dic:
        dic[s]+=1
    else:
        dic[s]=0

for i in dic:
    if dic[i]>1:
        cnt+=1
        li.append(i)

print(cnt)
li.sort()
for l in li:
    print(l)

File: 백준/Silver/1764. 듣보잡/듣보잡.java
import java.io.*;
import java.util.*;



public class Main {



    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st=new StringTokenizer(br.readLine());
        int n=Integer.parseInt(st.nextToken());
        int m= Integer.parseInt(st.nextToken());
        HashMap<String,Integer> h= new HashMap<>();
        ArrayList<String> ar= new ArrayList<>();
        int cnt=0;

        for(int i=0;i<n;i++){
            st=new StringTokenizer((br.readLine()));
            h.put(st.nextToken(),1);
        }
        for(int i=0;i<m;i++){
            st=new StringTokenizer((br.readLine()));
            String now=st.nextToken();
            int num = h.getOrDefault(now, 0);
            if(num>=1){
                ar.add(now);
                cnt+=1;
            }

        }

        System.out.println(cnt);

        Collections.sort(ar);
        for(int i=0;i<ar.size();i++){
            System.out.println(ar.get(i));
        }


    }

}